server/base/src/main/java/org/apache/accumulo/server/init/Initialize.java

```diff
@@ -644,6 +644,25 @@ private String getRootUserName(Opts opts) throws IOException {
     return rootpass.getBytes(UTF_8);
   }
 
+  /**
+   * Create warning message related to initial password, if appropriate.
+   *
+   * ACCUMULO-2907 Remove unnecessary security warning from console message unless its actually appropriate.
+   * The warning message should only be displayed when the value of <code>instance.security.authenticator</code>
+   * differs between the SiteConfiguration and the DefaultConfiguration values.
+   *
+   * @return String containing warning portion of console message.
+   */
+  private String getInitialPasswordWarning() {
+    StringBuilder optionalWarning = new StringBuilder();
+    Property authenticatorProperty = Property.INSTANCE_SECURITY_AUTHENTICATOR;
+    if (SiteConfiguration.getInstance().get(authenticatorProperty).equals(authenticatorProperty.getDefaultValue()))
+      optionalWarning.append(": ");
+    else
+      optionalWarning.append(" (this may not be applicable for your security setup): ");
+    return optionalWarning.toString();
```

Comment:
Can probably avoid use of StringBuilder here, since there is only a single string to return in each case, and it's not being concatenated with anything else until outside the method. Other than that, this change looks good to me. Thanks for the contribution!

Reply:
ACCUMULO-2907: Updated to replace StringBuilder usage with String.